app/assets/javascripts/admin/addon/components/modal/scrub-rejected-user.gjs

```diff
@@ -0,0 +1,67 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { isEmpty } from "@ember/utils";
+import DButton from "discourse/components/d-button";
+import DModal from "discourse/components/d-modal";
+import TextField from "discourse/components/text-field";
+import { i18n } from "discourse-i18n";
+
+export default class ScrubRejectedUserModal extends Component {
+  @tracked isScrubbing = false;
+  @tracked scrubReason = "";
+
+  @action
+  updateScrubReason(event) {
+    this.scrubReason = event.target.value;
+  }
+
+  @action
+  async confirmScrub() {
+    this.isScrubbing = true;
+    await this.args.model.confirmScrub(this.scrubReason);
+    this.args.closeModal();
+  }
+
+  get scrubButtonDisabled() {
+    return this.isScrubbing || isEmpty(this.scrubReason);
+  }
+
+  <template>
+    <DModal
+      @bodyClass="scrub-rejected-user"
+      class="admin-scrub-rejected-user-modal"
+      @title={{i18n "review.user.scrub_record.confirm_title"}}
+      @closeModal={{if this.isScrubbing null @closeModal}}
```

Comment:
NIT tracked `isScrubbing` property is false and here we check for null

Reply:
Isn't it returning `null` when `isScrubbing` is true? That's the intended behaviour, to prevent the modal from being closed whilst it's waiting for a server response.

Reply:
Sorry, I got confused, you are right :)